### PR TITLE
Add annotation for setting access mode on automounted configmap/secret files

### LINKED ISSUE #6549: Automatically mounting volumes, configmaps, and secrets
- [x] Implement specifications at https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#automatically-mounting-volumes-configmaps-and-secrets
- [x] Document behaviour (or point to the specification)
- [x] integration tests
- [x] Check with @kadel / @l0rd the new label replacing `controller.devfile.io/mount-to-devworkspace`
- [x] Only on cluster
- [ ] Implement controller.devfile.io/mount-access-mode (#6728)

related to #6478 epic

### LINKED ISSUE #6728: Add annotation for setting access mode on automounted configmap/secret files
/kind bug

https://github.com/devfile/devworkspace-operator/pull/1074

## What versions of software are you using?

**Operating System:**

**Output of `odo version`:**

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind feature

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6549 
Fixes #6728 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Review by rm3l — CHANGES_REQUESTED** [review]
Can you also add this new annotation to the related documentation page?
https://deploy-preview-6750--odo-docusaurus-preview.netlify.app/docs/user-guides/advanced/automounting-volumes/

File: docs/website/docs/user-guides/advanced/automounting-volumes.md [hunk 1]
```diff
@@ -37,3 +37,4 @@ Mounting resources can be additionally configured via annotations:
 
 - `devfile.io/read-only`: for persistent volume claims, mount the resource as read-only
 
+- `devfile.io/mount-access-mode`: for  secret/configmap, can be used to configure file permissions on mounted files
```
Comment (rm3l) [inline docs/website/docs/user-guides/advanced/automounting-volumes.md:40]: Can you also provide an example value? Along with the valid range?

This makes me wonder that we could add a complete example of an auto-mounting resource, with its labels and annotations. That would be helpful and easier to understand, IMO. But we can do that in a separate PR if you prefer..

Comment (feloy) [reply docs/website/docs/user-guides/advanced/automounting-volumes.md]: I've completed the doc, and added a test for decimal notation. 

I would prefer we discuss which kind of example we want and where to place it (blog, etc), and to create a separate PR for this.

**Review by rm3l — APPROVED** [review]

**Comment by feloy** [comment]
/override windows-integration-test/Windows-test
```
FAILED] [24.026 seconds]
odo devfile deploy command tests [AfterEach] when using a devfile.yaml containing an Image component with no build context should build image via Podman by defaulting build context to devfile path
  [AfterEach] 
```

**Comment by feloy** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests

Unrelated tests

```
[FAILED] [373.200 seconds]
odo describe/list binding command tests when creating a component with a binding as environment variables (service in namespace "") when Starting a Pg service when running dev session [BeforeEach] when changing the current namespace should list the binding with --namespace flag
  [BeforeEach] /go/odo_1/tests/integration/cmd_describe_list_binding_test.go:690
  [It] /go/odo_1/tests/integration/cmd_describe_list_binding_test.go:787
```

```
FAILED] [372.686 seconds]
odo describe/list binding command tests when creating a component with a binding as environment variables (service in namespace "binding-ncr") when Starting a Pg service when running dev session [BeforeEach] should list the binding
  [BeforeEach] /go/odo_1/tests/integration/cmd_describe_list_binding_test.go:690
  [It] /go/odo_1/tests/integration/cmd_describe_list_binding_test.go:742
```
